### ISSUE TITLE
Make the filewatcher change-checking more efficient

### ISSUE THREAD

**Issue by Webreaper** [Webreaper/Damselfly#271]
If the filewatcher triggers for a folder with 25,000 images in it, we shouldn't need to re-index all 25k images. Perhaps check the last-modified timestamp of the file and process them first, and then process the remaining ones after?

**Comment by Jogai** [comment]
Just process the files that are listed in the event.

https://docs.microsoft.com/en-us/dotnet/api/system.io.filesystemeventargs.fullpath?view=net-5.0#System_IO_FileSystemEventArgs_FullPath

**Comment by Webreaper** [comment]
Thanks. I already do that, but the filewatchers set up in Damselfly are set up per-folder, not per-file. Unfortunately, setting up per-file Filewatchers doesn't scale in .Net, particularly for large collections (my photo collection has 600,000 files; by the time you add in sidecars etc you'd end up with over a million filewatcher instances, which is massively memory-heavy and literally takes hours to initialise at startup (I know, I tried...).

**Comment by Jogai** [comment]
Yes I can see that. And I guess the folder watcher does not give you the changed files. That's a pity.
